book/website/reproducible-research/ci/ci-building-gh-actions.md

```diff
@@ -0,0 +1,72 @@
+# Building a Block of a Github Actions
+
+As described previously, workflow files use YAML syntax, which has either a `.yml` or `.yaml` file extension. If you're new to YAML and want to learn more, {ref}`see our section about YMAL<rr-renv-yaml>`. This workflow must be stored in the `.github/workflows` directory of your repository.
+
+Each workflow is defined in a separate YAML. We will introduce the building block of a workflow using Hello World Example:
+
+```
+name:
+    Hello World package
+on:
+  push:
+    branches: [ master ]
+Jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+```  
+
+**1. name**
+
+This is the name of the workflow and it is optional. GitHub will use this name to be displayed on the repository's actions page.
+```
+name:
+    Hello World package
+```
+
+**2. on**
+
+The `on` field tells GHA when to run. For example, we can run the workflow anytime there's a `push` or a `pull`.
+```
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+```
+There are many events which can be used to trigger a workflow. You can explore them [here](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions).
```

Comment:
overall it's good practice to describe what you would find in a link rather than using generic words, e.g. `here`, `click` etc. Now, the book has quite a few `here`, so it's not a big deal in this PR either. (I'm very much nit picking, I'm sorry).

```suggestion
There are many events which can be used to trigger a workflow. You can explore them [in the official GHA reference documentation](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions).
```